src/base/OpenMCCellAverageProblem.C

```diff
@@ -3116,4 +3134,50 @@ OpenMCCellAverageProblem::reloadDAGMC()
 #endif
 }
 
+void
+OpenMCCellAverageProblem::updateMaterials()
+{
+#ifdef ENABLE_DAGMC
+  // We currently only re-init the materials one time, because we create one new
+  // material for every density bin, even if that density bin doesn't actually
+  // appear in the problem. TODO: we could probably reduce memory usage
+  // if we only re-generated materials we strictly needed for the model.
+  if (!_first_transfer)
+    return;
+
+  // only need to create new materials if we have density skinning
+  if (_skinner->nDensityBins() == 1)
```

Comment:
Will the skinner try to assign a material name like `water_0` even with one density bin?

Reply:
No, the materials will not be changed if there is a single density bin (we `return` just after this line, in effect not modifying anything about the names). We _could_ change the names though if we wanted - I just thought it made more sense to skip that if density skinning was effectively not doing anything.

Reply:
I think that's okay for now. Just making a mental note. Thanks!